pkg/operator/genericoperatorclient/dynamic_operator_client.go

```diff
@@ -282,7 +282,11 @@ func (c dynamicOperatorClient) applyOperatorStatus(ctx context.Context, fieldMan
 	uncastOriginal, err := c.informer.Lister().Get(c.configName)
 	switch {
 	case apierrors.IsNotFound(err):
-		// do nothing and proceed with the apply
+		// set last transitionTimes and then apply
+		// If our cache improperly 404's (the lister wasn't synchronized), then we will improperly reset all the last transition times.
+		// This isn't ideal, but we shouldn't hit this case unless a loop isn't waiting for HasSynced.
+		v1helpers.SetApplyConditionsLastTransitionTime(&desiredConfiguration.Conditions, nil)
```

Comment:
Since `desiredConfiguration` can be `nil` (as per new check aded in line 313),  this can also panic, right?